From a29532948a78155d0bd750bacf44b964e77d1942 Mon Sep 17 00:00:00 2001
From: Chuck Lever <cel@kernel.org>
Date: Sun, 12 Jul 2026 16:45:49 -0400
Subject: [PATCH 12757/15333] NFSD: Remove '#include "nfsd.h"' from
 fs/nfsd/cache.h

Clean up: cache.h does not need the full definition of struct
nfsd_net. A forward declaration suffices.

Link: https://patch.msgid.link/20260712204554.125308-5-cel@kernel.org
Reviewed-by: Jeff Layton <jlayton@kernel.org>
Signed-off-by: Chuck Lever <cel@kernel.org>
---
 fs/nfsd/cache.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs/nfsd/cache.h b/fs/nfsd/cache.h
index bb7addef4a31..3bc4856e34b8 100644
--- a/fs/nfsd/cache.h
+++ b/fs/nfsd/cache.h
@@ -10,7 +10,8 @@
 #define NFSCACHE_H
 
 #include <linux/sunrpc/svc.h>
-#include "nfsd.h"
+
+struct nfsd_net;
 
 /*
  * Representation of a reply cache entry.
-- 
2.55.0

